bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/handler/BigAssFanHandler.java

```diff
@@ -1141,18 +1158,21 @@ private boolean isConnected() {
          * Periodically validate the command connection to the device by executing a getversion command.
          */
         private void scheduleConnectionMonitorJob() {
-            if (connectionMonitorJob == null) {
-                logger.debug("Starting connection monitor job in {} seconds for {} at {}", CONNECTION_MONITOR_DELAY,
-                        thing.getUID(), ipAddress);
-                connectionMonitorJob = scheduler.scheduleWithFixedDelay(connectionMonitorRunnable,
-                        CONNECTION_MONITOR_DELAY, CONNECTION_MONITOR_FREQ, TimeUnit.SECONDS);
+            synchronized (this) {
```

Comment:
Since the entire method body is within the `synchronized` block, you could add the keyword to the method signature instead. This would also ensure exact lock on this method, avoiding blocking and potentially deadlocking if other methods would also synchronize on `this`.